docker/pulsar/pom.xml

```diff
@@ -103,57 +107,71 @@
             </executions>
           </plugin>
           <plugin>
-            <groupId>com.spotify</groupId>
-            <artifactId>dockerfile-maven-plugin</artifactId>
-            <version>${dockerfile-maven.version}</version>
+            <groupId>io.fabric8</groupId>
+            <artifactId>docker-maven-plugin</artifactId>
             <executions>
               <execution>
                 <id>default</id>
+                <phase>package</phase>
                 <goals>
                   <goal>build</goal>
                 </goals>
-              </execution>
-              <execution>
-                <id>add-no-repo-and-version</id>
-                <goals>
-                  <goal>tag</goal>
-                </goals>
                 <configuration>
-                  <repository>pulsar</repository>
-                  <tag>${project.version}</tag>
+                  <images>
+                    <image>
+                      <name>${docker.organization}/pulsar</name>
+                      <build>
+                        <contextDir>${project.basedir}</contextDir>
+                        <tags>
+                          <tag>latest</tag>
+                          <tag>${project.version}</tag>
+                        </tags>
+                      </build>
+                    </image>
+                  </images>
                 </configuration>
               </execution>
               <execution>
-                <id>add-no-repo-and-latest</id>
+                <id>push-latest</id>
```

Comment:
By the way, I notice that there are redundant configurations here, and I think we can simplify them.

```xml
<executions>
      <execution>
        <id>default</id>
        <phase>package</phase>
        <goals>
          <goal>build</goal>
          <goal>push</goal>
        </goals>
        <configuration>
          <images>
            <image>
              <name>${docker.organization}/pulsar</name>
              <build>
                <contextDir>${project.basedir}</contextDir>
                <tags>
                  <tag>latest</tag>
                  <tag>${project.version}</tag>
                </tags>
              </build>
            </image>
          </images>
        </configuration>
      </execution>
</executions>
```

Then maybe we also need to update the https://github.com/apache/pulsar/blob/master/docker/publish.sh#L65

So like using `${docker.organization}/pulsar:latest` instead of `pulsar:latest`.

Reply:
There needs the PMC confirmation.

Reply:
This seems reasonable. I'm glad to integrate with this change if @sijie @merlimat @codelipenghui can confirm.

I found the original logic was introduced at https://github.com/apache/pulsar/pull/4705

Reply:
We don't need to change the image publish process

Please see: https://github.com/apache/pulsar/blob/master/wiki/release/release-process.md#10-publish-docker-images

Reply:
@codelipenghui I checked this process. Using the above config, we need to change the docker tag in `publish.sh`.

Reply:
@nodece I think the point here is that it's possible that the release manager doesn't have the permission to push images to apachepulasr org. Thus we leave the ability to push to another org and ask the maintainer to sync. In this case, we generate the image with no repository for adding org when `publish.sh`.

Based on this information, I tend to keep the logic as is and if you have further improvement idea, we can start a new thread to discuss it.

Reply:
Ok, we can make a new PR to improve that.